### PR TITLE
Add common TLS/SSL connection string options

### PR DESCRIPTION
**Pull request by ehlertjd**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -444,7 +453,7 @@ func ParseURL(url string) (*DialInfo, error) {
 			}
 			fallthrough
 		default:
-			return nil, errors.New("unsupported connection URL option: " + opt.key + "=" + opt.value)
+			fmt.Println("WARN: unsupported connection URL option: " + opt.key + "=" + opt.value)
```
Comment (roycollins) [inline session.go]: Just curious why the change in api behavior? If an invalid operation is provided why not keep the error? Is there a client that is sending in an invalid option that can not be corrected? 
This changes the current API response so less chance of it getting into upstream. Also it seems this might hide more issues like a mistyped case in the url.

**Review by roycollins — APPROVED** [review]
Seems solid to me. Just one question on changing behavior

File: session.go (hunk 1 above)
Comment (ehlertjd) [reply session.go]: No intention of getting this into the upstream - this behavior is more of an opinion that having an unknown query parameter in your connection string shouldn't crash or prevent the program from starting.

**Review by kofalt — CHANGES_REQUESTED** [review]
Supporting "tls" and changing the reject behavior strike me as opinions rather than necessary. I don't think that's appropriate for security-critical changes to components in maintenance mode. 

Similarly, can you justify both allowing bad SSL connections and loading CAs from a file in the context of needed operations? I assume the former makes sense for CI / mini-FW, but it would be nice to see that reasoning in the PR description.

Overall I think this should be reduced to the bare minimum patch that is needed to continue using this component until its replacement by the first party driver.

**Comment by ehlertjd** [comment]
@kofalt This has been pared down to the minimum necessary changes.

**Review by kofalt — APPROVED** [review]
